src/hotspot/share/memory/arena.cpp

```diff
@@ -366,6 +366,10 @@ void* Arena::grow(size_t x, AllocFailType alloc_failmode) {
 // Reallocate storage in Arena.
 void *Arena::Arealloc(void* old_ptr, size_t old_size, size_t new_size, AllocFailType alloc_failmode) {
   if (new_size == 0) return NULL;
```

Comment:
[pre-existing] Isn't this a leak?  Probably just dropping old_ptr.

Reply:
Yes, it is a leak, at least if we run with UseMallocOnly. I think this, and Amalloc, should behave consistently with os::malloc here, which is to return a small non-null allocation for size=0. Basically, if size==0 size=1.